domain/agentbinary/state/state_test.go

```diff
@@ -0,0 +1,205 @@
+
+	err := s.state.Add(context.Background(), agentbinary.Metadata{
+		Version:         "2.9.0",
+		Arch:            "amd64",
+		ObjectStoreUUID: objStoreUUID2,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	record := s.getAgentBinaryRecord(c, "2.9.0", archID)
+	c.Assert(record.Version, gc.Equals, "2.9.0")
+	c.Assert(record.ArchitectureID, gc.Equals, archID)
+	c.Assert(record.ObjectStoreUUID, gc.Equals, string(objStoreUUID2))
+}
+
+// TestAddErrorArchitectureNotFound asserts that a NotSupported error is returned
+// when the architecture is not found.
+func (s *stateSuite) TestAddErrorArchitectureNotFound(c *gc.C) {
+	objStoreUUID := s.addObjectStoreMetadata(c)
+
+	err := s.state.Add(context.Background(), agentbinary.Metadata{
+		Version:         "2.9.0",
+		Arch:            "non-existent-arch",
+		ObjectStoreUUID: objStoreUUID,
+	})
+	c.Assert(err, jc.ErrorIs, coreerrors.NotSupported)
+}
+
+// TestAddErrorObjectStoreUUIDNotFound asserts that a NotFound error is returned
+// when the object store UUID is not found.
+func (s *stateSuite) TestAddErrorObjectStoreUUIDNotFound(c *gc.C) {
+	s.addArchitecture(c, "amd64")
+
+	err := s.state.Add(context.Background(), agentbinary.Metadata{
+		Version:         "2.9.0",
+		Arch:            "amd64",
+		ObjectStoreUUID: objectstore.UUID(uuid.MustNewUUID().String()),
+	})
+	c.Assert(err, jc.ErrorIs, coreerrors.NotFound)
+}
+
+// Helper methods
+
+func (s *stateSuite) addArchitecture(c *gc.C, name string) int {
+	runner := s.TxnRunner()
+
+	// First check if the architecture already exists
+	selectStmt, err := sqlair.Prepare(`
+		SELECT id AS &architectureRecord.id
+		FROM architecture
+		WHERE name = $architectureRecord.name
+	`, architectureRecord{})
+	c.Assert(err, jc.ErrorIsNil)
+
+	record := architectureRecord{Name: name}
+	err = runner.Txn(context.Background(), func(ctx context.Context, tx *sqlair.TX) error {
+		return tx.Query(ctx, selectStmt, record).Get(&record)
+	})
+
+	// If architecture exists, return its ID
+	if err == nil {
+		return record.ID
+	}
+
+	// Otherwise insert the new architecture
+	insertStmt, err := sqlair.Prepare(`
+		INSERT INTO architecture (name)
+		VALUES ($architectureRecord.name)
+		RETURNING id AS &architectureRecord.id
+	`, architectureRecord{})
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = runner.Txn(context.Background(), func(ctx context.Context, tx *sqlair.TX) error {
+		return tx.Query(ctx, insertStmt, record).Get(&record)
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	return record.ID
+}
+
+func (s *stateSuite) addObjectStoreMetadata(c *gc.C) objectstore.UUID {
+	runner := s.TxnRunner()
+
+	storeUUID := uuid.MustNewUUID().String()
+	stmt, err := sqlair.Prepare(`
+		INSERT INTO object_store_metadata (uuid, sha_256, sha_384, size)
+		VALUES ($objectStoreMeta.uuid, $objectStoreMeta.sha_256, $objectStoreMeta.sha_384, $objectStoreMeta.size)
+	`, objectStoreMeta{})
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Generate unique SHA values using UUIDs
+	sha256Val := "sha256-" + uuid.MustNewUUID().String()
```

Comment:
UUIDs are still not real SHA.
You can just use a hardcoded string.

Reply:
Done & Done